chore: Bump required rust-toolchain version

ESHARK22 opened the pull request:
Use a newer rust-toolchain nightly version as some dependencies may require a higher MSRV

Commit:
chore: Bump required rust-toolchain version

TestingPlant commented:
Should we allow `clippy::manual_midpoint`? `f32::midpoint` generates more instructions (https://rust.godbolt.org/z/53Ph5x979), and float underflow/overflow isn't an issue we need to handle in general.

Commit:
style: Suppress clippy::manual_midpoint warning in aabb.rs

See https://github.com/hyperion-mc/hyperion/pull/949#issuecomment-3358052637
"f32::midpoint generates more instructions, and float underflow/overflow isn't an issue we need to handle in general."

Commit:
style: Collapse nested if statements

Addresses the follwing clippy error:
'''
error: this `if` statement can be collapsed
  --> crates/hyperion-clap-macros/src/lib.rs:14:9
   |
14 | /         if attr.path().is_ident("command_permission") {
15 | |             if let Err(err) = attr.parse_nested_meta(|meta| {
16 | |                 if meta.path.is_ident("group") {
17 | |                     if let Ok(Lit::Str(lit)) = meta.value()?.parse::<Lit>() {
...  |
27 | |         }
   | |_________^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_if
   = note: `-D clippy::collapsible-if` implied by `-D clippy::style`
   = help: to override `-D clippy::style` add `#[allow(clippy::collapsible_if)]`
help: collapse nested if block
   |
14 ~         if attr.path().is_ident("command_permission")
15 ~             && let Err(err) = attr.parse_nested_meta(|meta| {
16 |                 if meta.path.is_ident("group") {
...
25 |                     .into();
26 ~             }
   |
'''

Commit:
style: Use std::io::Error::other for cleaner error mapping

Fixes clippy::io-other-error

Commit:
style: remove unnecessary let binding

Commit:
style: Use const function

Commit:
style: Use explicit raw address-of operators

Fixes clippy::borrow_as_ptr (https://rust-lang.github.io/rust-clippy/master/index.html#borrow_as_ptr)

Commit:
style: Allow manual abs diff

Commit:
Merge branch 'main' into rustcbump
